sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumerClient.cs

```diff
@@ -335,7 +544,25 @@ await foreach (EventData item in subscription.ChannelReader.EnumerateChannel(max
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public virtual Task CloseAsync(CancellationToken cancellationToken = default) => InnerConsumer.CloseAsync(cancellationToken);
+        public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
+        {
+            Closed = true;
+
+            try
+            {
+                await InnerConsumer.CloseAsync(cancellationToken).ConfigureAwait(false);
+
+                if (OwnsConnection)
+                {
+                    await Connection.CloseAsync().ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex) when (ex is TaskCanceledException || ex is OperationCanceledException)
+            {
+                Closed = false;
```

Comment:
Fair point.  I think the closest that we can get for now is checking the `InnerXXX` in that `catch` block.    We know that the only potential cancellation comes from closing that, though it still doesn't give us strong guarantees.   Worst case, the `InnerXXX` will refuse to do things because it is closed and will throw an appropriate exception.

That all said, this will be something that doesn't really hold once we remove partition affinity anyway (and need multiple short-lived inner clients), so it's probably not worth digging deep into.